stl/inc/ranges

```diff
@@ -534,6 +535,332 @@ namespace ranges {
         inline constexpr _Single_fn single;
     } // namespace views
 
+    // CLASS TEMPLATE ranges::iota_view
+    template <class _Ty>
+    using _Iota_diff_t = conditional_t<is_integral_v<_Ty>, conditional_t<(sizeof(_Ty) < sizeof(int)), int, long long>,
+        iter_difference_t<_Ty>>;
+
+    // clang-format off
+    template <class _Ty>
+    concept _Decrementable = incrementable<_Ty> && requires(_Ty __t) {
+        { --__t } -> same_as<_Ty&>;
+        { __t-- } -> same_as<_Ty>;
+    };
+
+    template <class _Ty>
+    concept _Advanceable = _Decrementable<_Ty> && totally_ordered<_Ty> &&
+        requires(_Ty __i, const _Ty __j, const _Iota_diff_t<_Ty> __n) {
+            { __i += __n } -> same_as<_Ty&>;
+            { __i -= __n } -> same_as<_Ty&>;
+            _Ty(__j + __n);
+            _Ty(__n + __j);
+            _Ty(__j - __n);
+            { __j - __j } -> convertible_to<_Iota_diff_t<_Ty>>;
+        };
+
+    // clang-format off
+    template <weakly_incrementable _Wi>
+        requires semiregular<_Wi>
+    struct _Ioterator {

```

Comment:
Since the iterator doesn't depend on the sentinel type, this allows `_Ioterator<T>` to be reused for both `iota_view<T, U>` and `iota_view<T, V>`. (See ["What Are SCARY Iterators?"](https://devblogs.microsoft.com/cppblog/what-are-scary-iterators/) for details on this practice.)

Reply:
That is aweome

Reply:
That acronym makes absolutely no sense. The letters are just randomly picked from the words!

Reply:
I mean its C++. Our acronyms are even worse than our defaults